pkg/policy/validate.go

```diff
@@ -49,6 +49,12 @@ func Validate(policy *kyverno.ClusterPolicy, client *dclient.Client, mock bool,
 		if jsonPatchOnPod(rule) {
 			log.Log.V(1).Info("warning: pods managed by workload controllers cannot be mutated using policies. Use the auto-gen feature or write policies that match pod controllers.")
 		}
+
+		// validate rule name number of characters
+		if len(rule.Name) > 63 {
```

Comment:
@NoSkillGirl - can we use OpenAPI [schema](https://github.com/kyverno/kyverno/blob/1dfcef1cc8e334182648dfdeb3c3ea513f0c68ad/definitions/crds/kyverno.io_clusterpolicies.yaml#L581-L585) to validate policy definition? So that we can align the validate logic in CLI with what we have in the webhook.

Reply:
I've added OpenAPI Schema for clusterpolicy to [swagger.doc](https://github.com/realshuting/kyverno/commit/ad599ebbba9b33adc576e1ec2d4a2f12afee8c75), you should be able to use it for policy validation.